File: challenges/linear-search/js/linearSearch.js
// const arrayToSearchThrough = []; // empty list
// for (let i = 1; i <= 1000; i++) { // pushes 1 . . 1,000 into list
//     arrayToSearchThrough.push(i);
// }

exports.linearSearch = function (valueToFind, arrayToSearchThrough) {
    let index = 0;
    for (let i = 0; i <= arrayToSearchThrough.length; i++) {
        if (valueToFind === arrayToSearchThrough[i]) {
            return index;
        }
        index++;
    }
    return undefined;
};

File: challenges/linear-search/python/linear_search.py
array_to_search_through = []
for number in range(1, 1001):
    array_to_search_through.append(number)


def linear_search(value_to_find, array_to_search_through):
    index = 0
    for value in array_to_search_through:
        if value == value_to_find:
            return index
        index += 1
    return None


def global_linear_search(value_to_find, array_to_search_through):
    index = 0
    indices = []
    for value in array_to_search_through:
        if value == value_to_find:
            indices.append(index)
        index += 1
    return indices
